lib/buffer.js

```diff
@@ -807,8 +807,8 @@ Buffer.prototype.toJSON = function() {
 
 
 function adjustOffset(offset, length) {
-  offset |= 0;
-  if (offset === 0) {
+  offset = Math.floor(+offset);
+  if (offset === 0 || Number.isNaN(offset)) {
```

Comment:
Wouldn't it be better if we had `Number.isInteger` here and drop the `Math.floor` in the previous line?